tools/ci_build/github/azure-pipelines/templates/react-native-ci.yml

```
@@ -266,42 +295,62 @@ stages:
         targetFolder: $(Build.SourcesDirectory)/js/react_native/e2e/android/app/libs
       displayName: Copy Android package to Android e2e test directory
 
-    - template: android-dump-logs-from-steps.yml
-      parameters:
-        steps:
-        - task: Gradle@3
-          inputs:
-            gradleWrapperFile: '$(Build.SourcesDirectory)/js/react_native/e2e/android/gradlew'
-            workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e/android'
-            options: '--stacktrace'
-            tasks: ':app:connectedDebugAndroidTest'
-            publishJUnitResults: true
-            testResultsFiles: '**/TEST-*.xml'
-            testRunTitle: 'React Native Android e2e Test results'
-            javaHomeOption: 'path'
-            jdkDirectory: '$(JAVA_HOME_11_X64)'
-            sonarQubeRunAnalysis: false
-            spotBugsAnalysis: false
-          displayName: Run React Native Android e2e Tests
+    - script: |
+        detox build --configuration android.emu.release
+      workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+      displayName: Build React Native Detox Android e2e Tests
 
     - script: |
-        export FORCE_BUNDLING=1
-        export RCT_NO_LAUNCH_PACKAGER=1
-        export ENTRY_FILE=index.tsx
-        xcrun xcodebuild test -workspace $(Build.SourcesDirectory)/js/react_native/e2e/ios/OnnxruntimeModuleExample.xcworkspace \
-          -scheme OnnxruntimeModuleExample -destination 'platform=iOS Simulator,OS=latest,name=iPhoneRNTest' \
-          -derivedDataPath $(Build.BinariesDirectory)/react_native/ios_e2e_test/derived_data | xcpretty -r junit --no-color \
-          --output $(Build.SourcesDirectory)/js/react_native/e2e/ios/build/reports/test-results.xml
+        JEST_JUNIT_OUTPUT_FILE=$(Build.SourcesDirectory)/js/react_native/e2e/android-test-results.xml
+      workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+      displayName: Set E2E Jest Junit android test results output full path
+  
+    - script: |
+        detox test --record-logs all  --configuration android.emu.release
```

I think the previous step won't propagate the environment variable to the next as is. there's a way to output variables from a step (see [task.setvariable](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-variables-scripts?view=azure-devops&tabs=bash#about-tasksetvariable)), but it may be simpler to set the environment variable inline like:

```suggestion
        JEST_JUNIT_OUTPUT_FILE=$(Build.SourcesDirectory)/js/react_native/e2e/android-test-results.xml \
          detox test --record-logs all  --configuration android.emu.release
```

similar comment for the ios test